clang/lib/StaticAnalyzer/Frontend/CMakeLists.txt

```diff
@@ -21,6 +21,7 @@ add_clang_library(clangStaticAnalyzerFrontend
   clangCrossTU
   clangFrontend
   clangLex
+  clangSema
```

Comment:
Why this change is needed? Is this dependency introduced by SYCL changes?

Reply:
> Why this change is needed? 

Some other failures were hidden by the one that I fixed with the change in llvm/lib/Passes/CMakeLists.txt. I.e. after I fixed one `unresolved external` problem there were 4 more.

> Is this dependency introduced by SYCL changes?

Yes, I believe this failure was triggered by the recent pulldown to SYCL.
I'm not familiar with this part of FE, so I let FE people decide whether this dependency makes sense.

Reply:
This looks very suspicious. Community version doesn't have this dependency. https://github.com/llvm/llvm-project/blob/main/clang/lib/CodeGen/CMakeLists.txt#L92
+ @erichkeane, @AaronBallman, do you know the reason for clang's CodeGen to depend on Sema library?

Reply:
I'm going to merge this PR, but I'll open a separate issue to understand what's going on here.

Reply:
I think Clang's CodeGen library relies on Sema because of: https://github.com/llvm/llvm-project/blob/main/clang/lib/CodeGen/CGDecl.cpp#L44 but I admit that this looks really suspicious to me. I could have sworn this is a layering violation (now Sema relies on CodeGen which relies on Sema). The patch that introduced the change was: https://github.com/llvm/llvm-project/commit/0d61cd25a6927a7700bfb1636faca1ef16c46428 and it had all of 15 minutes of review before landing, and after landing someone pointed out the layering violation but no rollback happened. I've commented on it in community.